src/epoch.rs

```diff
@@ -1815,68 +1847,49 @@ impl Epoch {
-            TimeScale::TDB => self.to_tdb_duration_since_j1900(),
-            TimeScale::UTC => self.to_utc_duration(),
-            // GPST and QZSST share the same properties
-            TimeScale::GPST | TimeScale::QZSST => {
-                self.to_gpst_duration() + GPST_REF_EPOCH.to_tai_duration()
+    /*
+        /// Makes a copy of self and sets the duration and time scale appropriately given the new duration
+        #[must_use]
+        pub fn set(&self, new_duration: Duration) -> Self {
+            match self.time_scale {
+                TimeScale::TAI => Self::from_tai_duration(new_duration),
+                TimeScale::TT => Self::from_tt_duration(new_duration),
+                TimeScale::ET => Self::from_et_duration(new_duration),
+                TimeScale::TDB => Self::from_tdb_duration(new_duration),
+                TimeScale::UTC => Self::from_utc_duration(new_duration),
+                // GPST and QZSST share the same properties
+                TimeScale::GPST | TimeScale::QZSST => Self::from_gpst_duration(new_duration),
+                TimeScale::GST => Self::from_gst_duration(new_duration),
+                TimeScale::BDT => Self::from_bdt_duration(new_duration),
             }
-            TimeScale::GST => self.to_gst_duration() + GST_REF_EPOCH.to_tai_duration(),
-            TimeScale::BDT => self.to_bdt_duration() + BDT_REF_EPOCH.to_tai_duration(),
         }
-    }
-
-    /// Makes a copy of self and sets the duration and time scale appropriately given the new duration
-    #[must_use]
-    pub fn set(&self, new_duration: Duration) -> Self {
-        match self.time_scale {
-            TimeScale::TAI => Self::from_tai_duration(new_duration),
-            TimeScale::TT => Self::from_tt_duration(new_duration),
-            TimeScale::ET => Self::from_et_duration(new_duration),
-            TimeScale::TDB => Self::from_tdb_duration(new_duration),
-            TimeScale::UTC => Self::from_utc_duration(new_duration),
-            // GPST and QZSST share the same properties
-            TimeScale::GPST | TimeScale::QZSST => Self::from_gpst_duration(new_duration),
-            TimeScale::GST => Self::from_gst_duration(new_duration),
-            TimeScale::BDT => Self::from_bdt_duration(new_duration),
-        }
-    }
-
+    */
     #[must_use]
     /// Returns the number of TAI seconds since J1900
     pub fn to_tai_seconds(&self) -> f64 {
-        self.duration_since_j1900_tai.to_seconds()
+        self.to_tai_duration().to_seconds()
     }
 
     #[must_use]
     /// Returns this time in a Duration past J1900 counted in TAI
-    pub const fn to_tai_duration(&self) -> Duration {
-        self.duration_since_j1900_tai
+    pub fn to_tai_duration(&self) -> Duration {
+        let mut dur = self.duration + self.time_scale.tai_reference_epoch().duration;
+        if self.time_scale.uses_leap_seconds() {
+            dur += self.leap_seconds(true).unwrap_or(0.0) * Unit::Second;
```

Comment:
Leap second compensation is clearly needed, If i do it manually, it works.
But the `leap_seconds(true)` is causing a stack overflow, this has never cause problems before ?

Reply:
because _to_tai_duration_ is called inside  _leaps_seconds_with_